onboard/nav/searcher.cpp

```diff
@@ -231,22 +231,29 @@ NavState Searcher::executeTurnToBall( Rover * phoebe )
 // If the rover finishes driving to the ball, it moves on to the next Waypoint.
 // If the rover is on course, it keeps driving to the ball.
 // Else, it turns back to face the ball.
-NavState Searcher::executeDriveToBall( Rover * phoebe )
+NavState Searcher::executeDriveToBall( Rover * phoebe, const rapidjson::Document& roverConfig )
 {
     if( !phoebe->roverStatus().tennisBall().found )
     {
         cerr << "Lost the tennis ball\n";
         return NavState::SearchFaceNorth;
     }
-    if( phoebe->roverStatus().obstacle().detected )
+
+    double cvThresh = roverConfig[ "computerVision" ][ "obstacleThresh" ].GetDouble();
+    
+    if( phoebe->roverStatus().obstacle().detected and
```

Comment:
This if statement can be made more readable. Probably through the use of functions. In the function explain why we are subtracting 2 as well since this is not intuitive. My only guess is for being within 2 meters of a TB but even this is a guess. Also, we now attempt to get within 1 meter of TBs since 2.01 meters is 0 points.